core/appmodule/module.go

```diff
@@ -52,6 +52,22 @@ type HasPrecommit interface {
 	Precommit(context.Context) error
 }
 
+// ResponsePreBlock represents the response from the PreBlock method.
+// It can modify consensus parameters in storage and signal the caller through the return value.
+// When it returns ConsensusParamsChanged=true, the caller must refresh the consensus parameter in the finalize context.
+// The new context (ctx) must be passed to all the other lifecycle methods.
+type ResponsePreBlock interface {
+	IsConsensusParamsChanged() bool
```

Comment:
is this in order to repopulate consensusparams for other modules ?

Reply:
yes